### DISCUSSION TITLE
Etcd WatchChan loop exit somehow

### DISCUSSION THREAD

**Discussion by aeskkkey** [etcd-io/etcd#15720] [General]
I have a goroutine code like below. It tracks some etcd config files.
```
	wc := cfg.client.Watch(context.Background(), masterKeyPrefix, clientv3.WithPrefix(), clientv3.WithPrevKV(), clientv3.WithRev(rev))
	for wr := range wc {
		for _, event := range wr.Events {
			
		}
	}
```
The goroutine is supposed to run forever. 
But when my production code run for a long time, over a year for example, this goroutine may somehow exits.
So I have to add a panic code behind the for loop code and restart the whole app to rewatch it.

In go.mod, go version is 1.14, etcd version is go.etcd.io/etcd v0.0.0-20201125193152-8a03d2e9614b.

Can anyone explain why the WatchChan exit.

The app is watching other keys as well, but only one or two etcd-watch gorountine exit while others are ok. This confus me.

**Comment by liangyuanpeng** [comment]
The first suggestion is update your etcd package version :)
